src/ui/components/AnimateBlock/AnimateBlock.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+import block from 'bem-cn-lite';
+import {Waypoint} from 'react-waypoint';
+
+import './AnimateBlock.scss';
+
+const b = block('dl-animate-block');
+
+export interface AnimateBlockProps {
+    className?: string;
+    delay?: number;
+    offset?: number;
+    children?: React.ReactNode;
+}
+
+export const AnimateBlock = ({className, delay = 0, offset = 100, children}: AnimateBlockProps) => {
+    const [playAnimation, setPlayAnimation] = React.useState<boolean>(false);
+
+    return (
+        <div className={b({animate: playAnimation}, className)}>
+            <Waypoint
+                // trigger animation if element is above screen
+                topOffset={'-100000%'}
+                bottomOffset={offset}
+                onEnter={() => {
+                    setTimeout(() => {
```

Comment:
I don't know why but I texted comment here. Anyway `setTimeout` is side effect. It should be inside the useEffect hook with clearTimeout

Reply:
fixed – https://github.com/datalens-tech/datalens-ui/pull/707/commits/8c2d442732a89c508022554433918ea75e3fafd7